integration/cdi/src/main/java/org/apache/cxf/cdi/CdiClassUnwrapper.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.cdi;
+
+import java.util.List;
+import static java.util.Arrays.asList;
+import org.apache.cxf.common.util.ClassUnwrapper;
+
+class CdiClassUnwrapper implements ClassUnwrapper {
+    private static final List<String> UNWRAPPED_MARKERS = asList("OwbNormalScopeProxy", "WeldClientProxy");
```

Comment:
OWB has other suffixes, why not using $$ as in https://github.com/apache/meecrowave/blob/7bd58d4a254c0d40fb1a2ab0d19b24dbab4014ca/meecrowave-core/src/main/java/org/apache/meecrowave/cxf/MeecrowaveBus.java#L54 ? Would also handle TomEE proxies (EJB) BTW. Also any way it just gets merge with the default unwrapper? I don't see any reason to not have it out of the box since it doesn't depend on anything.

Reply:
I have only seen the one suffix from OWB.  Where can I find the list of suffixes?  I saw what you did in TomEE, and I think using `$$` could be problematic for non-CDI frameworks, so I wanted to explicitly list out the known suffixes.

I'm fine with putting something like this in core, and explicitly activating it in CDI, but would be concerned with being over eager with the swap out.

Reply:
For owb you need to check AbstractProxyFactory subclasses out, without more checking I had in mind $$OwbSubClass. In TomEE we also have $$LocalBeanProxy which would break probably, and a lot of frameworks use $$ for that purpose. That is why we use $$ and not something more precise which will always hit a pitfall. Do you know any framework using $$ and would break cxf unwrapping the class? If so we can surely just make the suffixes configurable. Throughts?

Reply:
@johnament @rmannibucau Was going through this discussion (https://issues.jboss.org/browse/CDI-10), seems to be a known problem people struggle with. Not sure if it is better or not, what if we detect the CDI implementation at runtime (class probing) and delegate the unwrapping to the framework itself? F.e. in case of OWB, we could use`NormalScopeProxyFactory.unwrapInstance(o)`, something similar should be available for Weld as well. Not ideal either, but more reliable may be? Thoughts?

Reply:
Well, I don't think Weld has a behavior like `unwrapInstance` but it could be done by us if its Weld.  I also figured that checking by name would be easier since it avoid any compile time dependencies.

Reply:
It is definitely the easiest / simplest way to do that, not sure how reliable it is though ... May be we could have some regex, like `^.+$$.+Proxy$`, the `$$` usually depict a generated class, so we would cover `Xxx$$OwbNormalScopeProxy`, `Xxx$$WeldClientProxy` as well as `Xxx$$LocalBeanProxy` like @rmannibucau  pointed out as well. Should be pretty good heuristic, what do you think?